test/serialize.js

```diff
@@ -25,3 +25,26 @@ test('should not have null byte if no body', function() {
     assert.deepEqual(actual, expected);
 });
 
+test('should not pass undefined or null through', function() {
+    var msg = new Msgs.NewOrderSingle();
+
+    msg.OrderQty = undefined;
+    msg.Price = null;
+
+    var out = msg.serialize();
+
+    var expected = [
+        '8=FIX.4.2',
+        '9=57',
+        '35=D',
+        '52=undefined',
```

Comment:
Not sure that the string undefined should appear in the expected value, right? It should just not exist in the body.